pkg/daemon/daemon.go

```diff
@@ -841,30 +852,67 @@ func (dn *Daemon) validateOnDiskState(currentConfig *mcfgv1.MachineConfig) bool
 
 // isUnspecifiedOS says whether an osImageURL is "unspecified",
 // i.e. we should not try to change the current state.
-func (dn *Daemon) isUnspecifiedOS(osImageURL string) bool {
+func isUnspecifiedOS(osImageURL string) bool {
 	// The ://dummy syntax is legacy
 	return osImageURL == "" || osImageURL == "://dummy"
 }
 
+// compareOSImageURL is the backend for checkOS.
+func compareOSImageURL(current, desired string) (bool, error) {
```

Comment:
Could we somehow add the details in the comments about what Canonical is from here(https://github.com/containers/image/blob/5144ced37a1b21b63c6ef605e56811e29a687528/docker/reference/reference.go#L133) so the logic is more clear in the func (ie so I don't have to click around to find it :smile: )